packages/react-components/lib/robots/mutex-group-table.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  DataGrid,
+  GridColDef,
+  GridEventListener,
+  GridValueGetterParams,
+  MuiEvent,
+  GridRowParams,
+} from '@mui/x-data-grid';
+import { useMediaQuery } from '@mui/material';
+import * as React from 'react';
+
+export interface MutexGroupData {
+  name: string;
+  lockedBy?: string;
+  requestedBy: string[];
+}
+
+export interface MutexGroupTableProps {
+  onMutexGroupClick?(
+    ev: MuiEvent<React.MouseEvent<HTMLElement>>,
+    mutexGroups: MutexGroupData,
+  ): void;
+  mutexGroups: MutexGroupData[];
+}
+
+export function MutexGroupTable({
+  onMutexGroupClick,
+  mutexGroups,
+}: MutexGroupTableProps): JSX.Element {
+  const isScreenHeightLessThan800 = useMediaQuery('(max-height:800px)');
+
+  const handleEvent: GridEventListener<'rowClick'> = (
+    params: GridRowParams,
+    event: MuiEvent<React.MouseEvent<HTMLElement>>,
+  ) => {
+    if (onMutexGroupClick) {
+      onMutexGroupClick(event, params.row);
+    }
+  };
+
+  const columns: GridColDef[] = [
+    {
+      field: 'mutexGroup',
+      headerName: 'Group',
+      maxWidth: 240,
+      editable: false,
+      valueGetter: (params: GridValueGetterParams) => params.row.name,
+      flex: 1,
+      filterable: true,
+    },
+    {
+      field: 'lockedBy',
+      headerName: 'Locked',
+      maxWidth: 240,
+      valueGetter: (params: GridValueGetterParams) => params.row.lockedBy ?? 'n/a',
+      flex: 1,
+      filterable: true,
+    },
+    {
+      field: 'requestedBy',
+      headerName: 'Waiting',
+      editable: false,
+      valueGetter: (params: GridValueGetterParams) => params.row.requestedBy?.join(', ') ?? 'n/a',
+      flex: 1,
+      filterable: true,
+    },
+  ];
+
+  return (
+    <DataGrid
```

Comment:
Can we enable `disableVirtualization` (https://mui.com/x/react-data-grid/virtualization/#disable-virtualization) to see if there will be any noticeable impact?

The reason is because the virtualization causes problem in tests after we upgrade the deps. Virtualization itself is also counter productive for anything other than huge tables. https://github.com/mui/mui-x/issues/1781

Reply:
https://github.com/open-rmf/rmf-web/pull/933/commits/8da155b9f1ae8f1a248eba852ba82be9d0061468